src/Storages/StorageS3.cpp

```diff
@@ -712,11 +715,13 @@ Chunk StorageS3Source::generate()
             UInt64 num_rows = chunk.getNumRows();
 
             const auto & file_path = reader.getPath();
-            size_t total_size = file_iterator->getTotalSize();
-            if (num_rows && total_size)
```

Comment:
may be also delete this getTotalSize method from iterator as it is no longer needed? afair it was added for this case

Reply:
Sure, I plan to delete it later in the next PR with using total_bytes_to_read instead of approx total_rows_to_read. Actually we will still need total size, but I will implement it in a different way